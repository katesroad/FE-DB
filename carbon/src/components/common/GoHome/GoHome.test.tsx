import * as React from 'react'
import { render, screen, within } from '@testing-library/react'
import { GoHome } from './index'
import { MemoryRouter } from 'react-router'

test('render <GoHome /> without providing css  class name', () => {
  const { container } = render(<GoHome />, { wrapper: MemoryRouter })
  expect(screen.getByRole('link')).toHaveAttribute('href', '/')
  expect(container).toMatchSnapshot()
})

test('render <GoHome /> css  class name', () => {
  const className = 'btn'
  const { container } = render(<GoHome className={className} />, {
    wrapper: MemoryRouter,
  })
  expect(within(container).getByRole('link').classList.contains(className))
})
