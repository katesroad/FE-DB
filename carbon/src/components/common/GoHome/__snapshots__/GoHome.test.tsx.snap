// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GoHome /> without providing css  class name 1`] = `
<div>
  <a
    class="ui-go-home"
    href="/"
  >
    <span>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 19l-7-7m0 0l7-7m-7 7h18"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <span>
        Go To Home
      </span>
    </span>
  </a>
</div>
`;
